ocean_provider/validation/RBAC.py

```diff
@@ -0,0 +1,145 @@
+            "DownloadRequest": "access",
+            "ComputeRequest": "compute",
+            "ComputeStartRequest": "compute",
+        }
+
+    def fails(self):
+        return False
+
+    def get_request_dict(self):
+        return json.loads(self.request["document"])
+
+    def get_dids(self, service_index: int):
+        req_dict = self.get_request_dict()
+        assets = list()
+        assets.append(req_dict["documentId"])
+        return [{"did": asset_did, "serviceId": service_index} for asset_did in assets]
+
+    def get_algos(self):
+        req_dict = self.get_request_dict()
+        algorithms = list()
+        algorithms.append(req_dict["algorithmDid"])
+        return [
+            {
+                "did": algorithm_did,
+                "serviceId": ServiceTypesIndices.DEFAULT_COMPUTING_INDEX,
+            }
+            for algorithm_did in algorithms
+        ]
+
+    def get_additional_dids(self):
+        req_dict = self.get_request_dict()
+        additional_inputs = (
+            req_dict["additionalInputs"] if "additionalInputs" in req_dict else []
+        )
+        return [
+            {
+                "did": additional_input["documentId"],
+                "serviceId": additional_input["serviceId"],
+            }
+            for additional_input in additional_inputs
+        ]
+
+    def build_payload(self):
+        payload = {
+            "eventType": self.action,
+            "component": self.component,
+            "providerAddress": self.provider_address,
+            "credentials": self.credentials,
+        }
+        # builds actions like build_encrtyptUrl_payload to update the dictionary
+        # with request - specific key-values.
+        payload.update(getattr(self, f"build_{self.action}_payload")())
+        return payload
+
+    def build_encryptUrl_payload(self):
+        message = "encryptUrl" + json.dumps(self.credentials)
+        signature = sign_hash(msg_hash(message), get_provider_wallet())
+
+        return {"signature": signature}
+
+    def build_initialize_payload(self):
+        dids = self.get_dids(ServiceTypesIndices.DEFAULT_ACCESS_INDEX)
+        message = "initialize" + json.dumps(self.credentials)
+        signature = sign_hash(msg_hash(message), get_provider_wallet())
+        return {
+            "signature": signature,
+            "dids": dids,
+        }
+
+    def build_access_payload(self):
+        dids = self.get_dids(ServiceTypesIndices.DEFAULT_ACCESS_INDEX)
+        message = "access" + json.dumps(self.credentials)
+        signature = sign_hash(msg_hash(message), get_provider_wallet())
+        return {
+            "signature": signature,
+            "dids": dids,
+        }
+
+    def build_compute_payload(self):
+        dids = self.get_dids(ServiceTypesIndices.DEFAULT_COMPUTING_INDEX)
+        algos = self.get_algos()
+        additional_dids = (
+            self.get_additional_dids() if self.get_additional_dids() else []
```

Comment:
Please add the if inside the function itself, to keep things cleaner. E.g. the first statement inside the get_additional_dids function can be precisely to return [] if nothing exists